File: CS/DXSample/GridViewModel.cs
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace DXSample {
    public class GridViewModel : ViewModelBase {
        public ObservableCollection<ItemViewModel> Items {
            get { return GetValue<ObservableCollection<ItemViewModel>>(); }
            set { SetValue(value); }
        }
        public GridViewModel() {
            Items = new ObservableCollection<ItemViewModel>();
            for (int i = 0; i < 10; i++) {
                Items.Add(new ItemViewModel() { Subject = $"Mail{i}", IsRead = i % 3 == 0 });
            }
        }
    }

    public class ItemViewModel : ViewModelBase {
        public string Subject {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public bool IsRead {
            get { return GetValue<bool>(); }
            set { SetValue(value); }
        }
    }
}

File: VB/DXSample/GridViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace DXSample

    Public Class GridViewModel
        Inherits ViewModelBase

        Public Property Items As ObservableCollection(Of ItemViewModel)
            Get
                Return GetValue(Of ObservableCollection(Of ItemViewModel))()
            End Get

            Set(ByVal value As ObservableCollection(Of ItemViewModel))
                SetValue(value)
            End Set
        End Property

        Public Sub New()
            Items = New ObservableCollection(Of ItemViewModel)()
            For i As Integer = 0 To 10 - 1
                Items.Add(New ItemViewModel() With {.Subject = $"Mail{i}", .IsRead = i Mod 3 = 0})
            Next
        End Sub
    End Class

    Public Class ItemViewModel
        Inherits ViewModelBase

        Public Property Subject As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property IsRead As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
